server/hap/HapProcessZabbixAPI.cc

```diff
@@ -70,18 +70,27 @@ void HapProcessZabbixAPI::setMonitoringServerInfo(void)
 
 void HapProcessZabbixAPI::workOnTriggers(void)
 {
-	SmartTime lastTriggerTime = getTimestampOfLastTrigger();
-	// TODO: getTrigger() should accept SmartTime directly.
-	// TODO: We should add a way to get newly added triggers.
-	//       Their timestamp are 0 in UNIX time. So the following way
-	//       cannot retrieve them.
-	const int requestSince = lastTriggerTime.getAsTimespec().tv_sec;
-	ItemTablePtr triggers = getTrigger(requestSince);
-	ItemTablePtr expandedDescriptions =
-	  getTriggerExpandedDescription(requestSince);
-	ItemTablePtr mergedTriggers =
-	  mergePlainTriggersAndExpandedDescriptions(triggers, expandedDescriptions);
-	sendTable(HAPI_CMD_SEND_UPDATED_TRIGGERS, mergedTriggers);
+	HapiTriggerCollectType type = getTriggerCollectType();
+	if (type == DIFFERENCE_TRIGGER_COLLECT) {
```

Comment:
We can extract similar lines outside the condition blocks.
Of course, some parameters will be set in the blocks.
Regardless of whether all triggers are gotten, basic procedure is the same.